### PR TITLE
Add Rm module

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: src/Streamly/Coreutils/Rm.hs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+force :: Rm
+force = Force
+
+recursive :: Rm
+recursive = Recursive
+
+instance Default Rm
+    where
+    def = Force
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Rm.hs:36]: Defaulting to force remove?

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: streamly-coreutils.cabal [hunk 2]
```diff
@@ -97,6 +97,7 @@ library
       Streamly.Coreutils
     , Streamly.Coreutils.Common
     , Streamly.Coreutils.Cp
+     , Streamly.Coreutils.Rm
```
Comment (harendra-kumar) [inline streamly-coreutils.cabal:100]: Remove additional space.

File: src/Streamly/Coreutils/Rm.hs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+rmForce :: Switch -> Rm -> Rm
+rmForce sw options = options {force = sw}
+
+rmRecursive :: Switch -> Rm -> Rm
+rmRecursive sw options = options {recursive = sw}
+
+rm :: Rm -> FilePath -> IO ()
+rm opt = do
+    case opt of
+        Rm {force = On} -> removePathForcibly
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Rm.hs:45]: BUG: This will remove directories recursively even if `recursive` is not On.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: src/Streamly/Coreutils/Rm.hs (hunk 1 above)
Comment (rnjtranjan) [reply src/Streamly/Coreutils/Rm.hs]: Fixed.

File: src/Streamly/Coreutils/Rm.hs [hunk 4]
```diff
@@ -16,32 +16,37 @@ module Streamly.Coreutils.Rm
...
+rm f path = do
+    let opt = f defaultConfig
     case opt of
-        Rm {force = On} -> removePathForcibly
-        Rm {recursive = On} -> removeDirectoryRecursive
-        _ -> removeFile
+        Rm {force = On, recursive = On} -> removePathForcibly path
+        Rm {force = On, recursive = Off} -> do
+            notDir <- test path isFile
+            when notDir $ removePathForcibly path
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Rm.hs:50]: Can not use removePathForcibly without the `nuke` options, as explained above.

Here we should do:

* Check if the file exists, if does not exist then do nothing
* If it exists then call removeFile

File: src/Streamly/Coreutils/Rm.hs [hunk 5]
```diff
@@ -16,32 +16,37 @@ module Streamly.Coreutils.Rm
...
+        Rm {force = Off, recursive = On} -> removeDirectoryRecursive path
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Rm.hs:51]: It should give an error if the target directory does not exist.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by rnjtranjan — COMMENTED** [review]
looks good.

**Review by rnjtranjan — COMMENTED** [review]
looks fine.

**Pull request merged**